x-pack/plugin/ilm/src/internalClusterTest/java/org/elasticsearch/xpack/ilm/DataStreamAndIndexLifecycleMixingTests.java

```diff
@@ -598,20 +603,68 @@ public void testUpdateIndexTemplateToDataStreamLifecyclePreference() throws Exce
             assertThat(secondGenerationExplain.getPhase(), is("hot"));
             assertThat(secondGenerationExplain.getStep(), is(PhaseCompleteStep.NAME));
 
-            // ILM is not managing this index anymore
+            // ILM is not managing these indices anymore
             IndexLifecycleExplainResponse thirdGenerationExplain = explainResponse.getIndexResponses().get(thirdGenerationIndex);
             assertThat(thirdGenerationExplain.managedByILM(), is(false));
 
-            // the write index is managed by the data stream lifecycle
+            IndexLifecycleExplainResponse writeIndexExplain = explainResponse.getIndexResponses().get(writeIndex);
+            assertThat(writeIndexExplain.managedByILM(), is(false));
+
+            // the third generation and the write index are managed by the data stream lifecycle
             ExplainDataStreamLifecycleAction.Response dataStreamLifecycleExplainResponse = client().execute(
                 ExplainDataStreamLifecycleAction.INSTANCE,
                 new ExplainDataStreamLifecycleAction.Request(new String[] { thirdGenerationIndex, writeIndex })
             ).actionGet();
             assertThat(dataStreamLifecycleExplainResponse.getIndices().size(), is(2));
             for (ExplainIndexDataStreamLifecycle index : dataStreamLifecycleExplainResponse.getIndices()) {
                 assertThat(index.isManagedByLifecycle(), is(true));
+                assertThat(index.getLifecycle(), equalTo(customLifecycle));
             }
         });
+
+        // Then we opt out the data stream lifecycle
+        // let's migrate this data stream to use the data stream lifecycle starting with the next generation
+        client().execute(
+            PutDataStreamLifecycleAction.INSTANCE,
+            new PutDataStreamLifecycleAction.Request(new String[] { dataStreamName }, TimeValue.timeValueDays(90), false)
+        );
```

Comment:
Wouldn't this mean (if the comment is correct) that the data stream would stop being managed immediately, rather than only at the next generation? I think the comment is wrong, and should perhaps say:

```suggestion
        // Then we opt out the data stream lifecycle
        // let's migrate this data stream to disable using the data stream lifecycle
        client().execute(
            PutDataStreamLifecycleAction.INSTANCE,
            new PutDataStreamLifecycleAction.Request(new String[] { dataStreamName }, TimeValue.timeValueDays(90), false)
        );
```